components/eamxx/src/physics/ml_correction/eamxx_ml_correction_process_interface.cpp

```diff
@@ -135,6 +147,61 @@ void MLCorrection::run_impl(const double dt) {
       ML_model_tq, ML_model_uv, ML_model_sfc_fluxes, datetime_str);
   pybind11::gil_scoped_release no_gil;  
   ekat::enable_fpes(fpe_mask);   
+
+  // Now back out the qv change abd apply it to precipitation, only if Tq ML is turned on
+  if (not (m_ML_model_path_tq == "None")) {
+    using PC  = scream::physics::Constants<Real>;
+    using KT  = KokkosTypes<DefaultDevice>;
+    using MT  = typename KT::MemberType;
+    using ESU = ekat::ExeSpaceUtils<typename KT::ExeSpace>;
+    const auto &pseudo_density       = get_field_in("pseudo_density").get_view<const Real**>();
+    const auto &precip_liq_surf_mass = get_field_out("precip_liq_surf_mass").get_view<Real *, Host>();
+    const auto &precip_ice_surf_mass = get_field_out("precip_ice_surf_mass").get_view<Real *, Host>();
+    constexpr Real g = PC::gravit;
+    const auto num_levs = m_num_levs;
+    const auto policy = ESU::get_default_team_policy(m_num_cols, m_num_levs);
+    
+    const auto &qv_tnew = get_field_in("qv").get_view<const Real **>();
```

Comment:
What is the difference between `get_field_in` and `get_field_out`? Above we use the `get_field_out` to access the `qv` field.

Reply:
There is added protection for the `get_field_in` call such that it must be `const`.  In other words, a field can be gathered either by get_field_in or get_field_out within an atm. process.  With the following rules about when the field was registered w/ the process:

as Required: only get_field_in
as Computed: only_get_field_out
as Updated: both get_field_in and get_field_out

For protection though, if you are grabbing a field with no intention of changing it you use get_field_in, even if you could use get_field_out.

Reply:
Some more historical context: in the early SCREAM days, fields were templated on data type, which was `Real` for output fields and `const Real` for input fields. At that time, we had no choice but to store them separately, and therefore have two separate `get_field_in/out` methods. Now, fields are no longer templated on the data type (they handle read-only constraints at runtime), so we _could_ merge the `get_field_in/out` methods in one (and I thought about doing it).